engine/derivative.cc

```diff
@@ -639,6 +640,34 @@ namespace MathDAG
        throw error("frac is not differentiable");
   }
   
+  template <>
+  NodePtr SystemOfEquations::derivative<>
+  (const OperationDAG<OperationType::percent>& expr)
+  {
+    if (expr.arguments[0].empty())
+      return zero;
+    else
+      {
+        Expr x(expressionCache, expr.arguments[0][0]);
+        return chainRule(x, 100*x/x);
```

Comment:
The derivative of % is just 100x'

No need to invoke the chain rule, you can write this as 100*x->derivative(*this);

Reply:
Thanks, I fixed this.